### PR TITLE
lll: disallow useless PUSHn in assembly

### PR DESCRIPTION
**Pull request by axic**
Found in #3002.

### CODE REVIEW

File: test/liblll/Compiler.cpp [hunk 1]
```diff
@@ -121,6 +121,23 @@ BOOST_AUTO_TEST_CASE(switch_inconsistent_return_count)
...
+
+BOOST_AUTO_TEST_CASE(disallowed_functional_asm_instructions)
+{
+	for (unsigned i = 1; i <= 32; i++)
+		BOOST_CHECK(!successCompile("(PUSH" + boost::lexical_cast<string>(i) + ")"));
+	for (unsigned i = 1; i <= 16; i++)
+		BOOST_CHECK(!successCompile("(DUP" + boost::lexical_cast<string>(i) + ")"));
+	for (unsigned i = 1; i <= 16; i++)
+		BOOST_CHECK(!successCompile("(SWAP" + boost::lexical_cast<string>(i) + ")"));
+	BOOST_CHECK(!successCompile("(JUMPDEST)"));
```
Comment (pirapira) [inline test/liblll/Compiler.cpp:138]: @winsvega wanted to use `(JUMPDEST)` in some tests.  He and @axic need to talk.

Comment (axic) [reply test/liblll/Compiler.cpp:138]: It can still be used via assembly: `(asm JUMPDEST)`. See #3066.

Comment (pirapira) [reply test/liblll/Compiler.cpp:138]: OK, then it's a non-issue.

Comment (axic) [reply test/liblll/Compiler.cpp:138]: Also note it was removed a week ago or so, this PR removes `pushnn` from assembly and adds tests for the rest too.

Comment (winsvega) [reply test/liblll/Compiler.cpp:138]: it is an issue. some tests rely on (JUMPDEST) 
and we dont have a unit test that actually checks that all lll opcodes are actually supported by lllc version before running the test fillers.

Comment (axic) [reply test/liblll/Compiler.cpp:138]: In any case this pull request doesn't remove `(JUMPDEST)` as it is already removed. I'd vote for merging this pull request as it fixes another bug.

Comment (axic) [reply test/liblll/Compiler.cpp:138]: The problem with the tests is it uses LLL as if it were inline assembly, but it isn't. Using inline assembly in LLL (that is `(asm <instr>...)`) would be more correct or if functional assembly is needed, using Solidity's assembly.

Comment (winsvega) [reply test/liblll/Compiler.cpp:138]: tests use lllc as parser. just parser of (opcode) to bytecoded and thats it.

Comment (pirapira) [reply test/liblll/Compiler.cpp:138]: @winsvega now that is not LLL.  We need a new language.

Comment (winsvega) [reply test/liblll/Compiler.cpp:138]: @pirapira what are the options?

Comment (pirapira) [reply test/liblll/Compiler.cpp:138]: We build just parser of (opcode) to bytecode.

Comment (winsvega) [reply test/liblll/Compiler.cpp:138]: the older version of  solidity lllc works just like that. if you suggest to write a new parser...

Comment (axic) [reply test/liblll/Compiler.cpp:138]: The older version of lllc is buggy and it produces unreliable output, at least output which we shouldn't rely on in testing the entire ecosystem with.

Comment (pirapira) [reply test/liblll/Compiler.cpp:138]: @winsvega either fork an older lllc and fix the bugs, or create a new compiler.

**Comment by axic** [comment]
@pirapira @chriseth can we merge this?

**Comment by pirapira** [comment]
Yes, I think so when the tests pass.

**Comment by axic** [comment]
@pirapira the tests failures are not related to this PR

**Comment by pirapira** [comment]
Restarted the emscripten build.

**Comment by axic** [comment]
@pirapira ok to merge?

**Review by pirapira — APPROVED** [review]
Looks good.

**Pull request merged**